lib/lynx/config.rb

```diff
@@ -9,26 +9,25 @@ def initialize(config = {})
       @config = config
     end
 
-    [:username, :password, :host, :database, :socket].each do |method|
+    [:credentials, :username, :host, :database, :socket].each do |method|
       define_method(method){ self[method] }
     end
 
     def mysql
-      @mysql ||= self[:mysql] || detect(MYSQL)
+      @mysql ||= self[:mysql] || detect(MYSQL) || raise(Lynx::Error, '[ERROR] Failed to detect a valid version of mysql')
```

Comment:
We probably don't need  the `[ERROR]` tags  here since these are wrapped in exceptions now.